### ISSUE TITLE
[Bug] markArea xAxis does not work as expected for heatmap

### ISSUE THREAD

**Issue by melquelima** [apache/echarts#20387]
### Version

5.5.1

### Link to Minimal Reproduction

https://codesandbox.io/p/sandbox/magical-bose-sv8qm7?file=%2Fsrc%2Findex.js%3A98%2C2

### Steps to Reproduce

open link to preview the sample
https://codesandbox.io/p/sandbox/magical-bose-sv8qm7?file=%2Fsrc%2Findex.js%3A98%2C2

OR

1 - open heat map sample in echarts website
https://echarts.apache.org/examples/en/editor.html?c=heatmap-cartesian&version=5.5.1
2 - add Markarea code

`markArea: {
          data: [ [{name: 'mark',xAxis: '2a'}, { xAxis: '3a'}]],
          label: {
              color: 'green'
          },
          itemStyle: {
              color: 'orange',
          }
      }`

### Current Behavior

markarea in heatmap isnt working as spected

![image](https://github.com/user-attachments/assets/ac8254b1-bf98-46d0-aade-476515967163)

it works fine when it is in bar mode:

![image](https://github.com/user-attachments/assets/2665bc8c-1848-4d23-862c-d8016f87a403)

### Expected Behavior

fix this issue

### Environment

```markdown
- OS: Windows
- Browser: Chrome
- Framework: None
```

**Comment by helgasoft** [comment]
agree, would be better if _heatmap.markArea_ display is consistent with _bar.markArea_.
See if [changing axisTick](https://echarts.apache.org/examples/en/editor.html?c=heatmap-cartesian&version=5.5.1&code=PTAEAcCcFMBdYJbUgWgQcwHYHsYCgBjbTAZ1lAAtsBXSE0AXlAG09R3QByARgCYBDTgBou3QSM4DhXAMziuAFnmcArMoBs4th04B2ZQA5lATmXcADGbHC8AXQDceEBBjwkqDDnxFS5ACb8AJ70TKwcXADK_LC0AYHSnABikAhxCQAqFLQkaULa7JwA6tB-mNA5QRnU5blcALLEtZwR1Jhpdo7OUHCIyGhYuNCExGSgAbD8jCzM5iKzoCq2IjMi3KtLLPO8cxsroDI7y_MKh5siKqd76pfzujciBvegxk8Wr2ugvLvzfCIK36sDqBuADgSdgaDuBd9pDrqB_kdVndQOpIY94ZCXhjEZ95jJQbwPvicbxtp8CUDFssPrddh8aXSRGTzIz9pcPuCWdTzuyRHCuSwPsiBcwPuiRR8sRLVvMqYLVh8vtzgWSlfLgUDUcruODjKyoQr9XCteruMjdPr0QZ9Vi-KzeLL7R85aKmWSLcreEC1cxmetlmT6btVZcyUCRWTOcGeaAI3zQyJhdHQOLk1Lk28YQGFUyM4GM0C9dmwQ8M9CLBnjRnkS7A-ja6ssQ3cYnk4S_m3VW2gQiWOHW8sgUTdt7LkDwyO_mOYyLNdPQEnBw9_X2ROml8DHRvuBzJyqB6uNate8wh-DuMTDwaVXvuHCi1ea7f67fbbOmfMTaembuN6SO3-Q67JyK7MOeIiXmBTKXOCE7LOCUbwTGIJISiMGJuhKaYViPrnscwE5tiLDnl2qEXgBxHHhRUHAuWJ7nnCdpkciKGUcCVoEcCTacQ6EE8UKPFhjxcEsNCWy7OWoHQsyEkQXxyzQohokzrJaGxqpi7KVh5IKWulzlvhunAkGRm_AsqnkZuFngi65bQh6Wl3oapnIl-5apqZ3FGbxWZae2RHMNJpF-SJzD8qBjGXHCMnLHCoVwkpYUqbF8bqSlC5RcuaUsPemWbhFhEioxwVJUeELpTq8k5asim7Ix_J1UieXcB51VcXlPmsaV_k-tFJXRZqti2PkoAAHQALb8OAAAUABmrQEIgxCgNNCCwNA40AJSgAA3iNri0JgLBrRtoobCd40zOd62XV8oAAD73VwKCcA4eAAL6bY42DgEtR1MHt7CwNg2AADaIOAABcu0jeA2AkGtCDENDnDA-AnDaO9eTsOgKR-NDgMcBQ0AYBQsAoyo5gAKQ2OEaMoxYNOY9joAAB4AIKswgJAEyNsCBOA0AowQ0TQOguDxCz7DjPw0NUNkUugCQ4Cg2t7MwLLu3hOwJBUAA7tDsCQNU4RYyN_BcyQ6QIAQADWkM7fwqtYIUa0UAAMvwABG0Cg5DRvVO9zPaIEnPc7zdMC0LXAi-t4uQJLI0y9DcQkIryuq7A6vQJrhPhLr2AG6AAdDKbwfsAAbtz1BO3UU0Rxw40IJg0OzCNk2s9DbwjSLoMENQoPe6D0cl4ruBIJg5NcFQKQAF7EBMoO0xww-zVPnAENAk_IMv7Be9g8DYONDMqEz7BmzryBIDzLAjXnHCYPw43R5wAAKC0UKAADC_CQH4u8cH5oLFGxNoiTXRoraW0RNYy0gaAQePs_Yw21hwAuRcS4jXPnAja4AKD8ARjfe-4QLoRH5sPBuKCdZ4L8IXAAQqDWgXc26UNQdQwuX8wa4BRpAdAXt-DTXmIIuYo0VCbQxiwoOKCL7a0mpAW22dc6gEwRwZOLAlEsPYMwHaj9n4o1kbbaQHNLYoykFjXaRjw6KEEO9YaLCNjKPYAg32BNQBEFBlwrguNoBb04KAMxDjQAkLIdHHarjOGQBRuAZuBiRDvT8SNSRoBhrvXsEAA) could help in some way.
![image](https://github.com/user-attachments/assets/3843bfec-ff8e-4e86-a6ea-8f73c4571fd3)

**Comment by melquelima** [comment]
@helgasoft it makes a little confusion when in heatmap vew when Xaxis is not centered, like we dont know if value "3" belogns to 2a,3a or 4a

**Comment by melquelima** [comment]
no news?
